admin-data-explorer-play-server/src/main/scala/net/wiringbits/webapp/utils/admin/repositories/daos/DatabaseTablesDAO.scala

```diff
@@ -78,25 +79,47 @@ object DatabaseTablesDAO {
       queryParameters: QueryParameters,
       settings: TableSettings
   )(implicit conn: Connection): List[TableRow] = {
+    val datePattern = "([12]\\d{3})-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01])".r
     val limit = queryParameters.pagination.end - queryParameters.pagination.start
     val offset = queryParameters.pagination.start
     // react-admin gives us a "id" field instead of the primary key of the actual column so we need to replace it
     val sortBy = if (queryParameters.sort.field == "id") settings.primaryKeyField else queryParameters.sort.field
 
     val preparedStatement = queryParameters.filter.field map { field =>
       val primaryKey = if (field == "id") settings.primaryKeyField else field
-      val sql =
-        s"""
-      SELECT * FROM $tableName
-      WHERE $primaryKey = ?
-      ORDER BY $sortBy ${queryParameters.sort.ordering}
-      LIMIT ? OFFSET ?
-      """
-      val preparedStatement = conn.prepareStatement(sql)
-      preparedStatement.setString(1, queryParameters.filter.value)
-      preparedStatement.setInt(2, limit)
-      preparedStatement.setInt(3, offset)
-      preparedStatement
+      val filterValue = queryParameters.filter.value
+      filterValue match {
+        case datePattern(year, month, day) =>
+          val parsedDate = LocalDate.of(year.toInt, month.toInt, day.toInt)
+          val initDate = Date.valueOf(parsedDate)
+          val endDate = Date.valueOf(parsedDate.plusDays(1L))
+          val sql =
+            s"""
+          SELECT * FROM $tableName
+          WHERE $primaryKey BETWEEN ? AND ?
```

Comment:
This seems wrong to me, we are filtering data based on primary key which isn't a date.

Also, what about text/number filters?